### PR TITLE
Migrate app/react-native to typescript

### PR DESCRIPTION
**Pull request by benoitdion**
Issue: #5030

## What I did

Migrate app/react-native to typescript. Most types have been annotated properly except for the data we get from client-api.

### CODE REVIEW

File: app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx [hunk 1]
```diff
@@ -20,13 +19,19 @@ const style = StyleSheet.create({
...
 
-export default class Navigation extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      isUIVisible: props.initialUiVisible,
-    };
-  }
+interface Props {
+  initialUiVisible: boolean;
```
Comment (gaetanmaisse) [inline app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx]: According to https://github.com/storybooks/storybook/pull/6448/commits/ea0c8a0f4a997791ed4999c0f8d8b5cec487089a#diff-46b88ac54083f1f8e54a54e6f4a3088fL77
`initialUiVisible` is optional so maybe `initialUiVisible?: boolean;` is more accurate

File: app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import React, { PureComponent } from 'react';
 import { View, SafeAreaView, StyleSheet } from 'react-native';
+// @ts-ignore TODO: add types for react-native-swipe-gestures
```
Comment (gaetanmaisse) [inline app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx]: or add a typings.d.ts with `declare module 'react-native-swipe-gestures'` if types are not available for this lib 😉

**Review by gaetanmaisse — APPROVED** [review]

File: app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx (hunk 1 above)
Comment (benoitdion) [reply app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx]: good catch

File: app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx (hunk 2 above)
Comment (benoitdion) [reply app/react-native/src/preview/components/OnDeviceUI/navigation/index.tsx]: added in 29ddf5aafc483b9571cac5d4378ff53d6c4ae185

**Pull request merged**